### DISCUSSION TITLE
[Question] ESP Filesystem -> warning: 'SPIFFS' is deprecated: SPIFFS has been deprecated. Please consider moving to LittleFS

### DISCUSSION THREAD

**Discussion by hasenradball** [Bodmer/TFT_eSPI#2946] [Q&A - General]
Hi,

old stuff I know.
Do you know when you will switch to `LITTLEFS` Filesystem for ESP8266 and ESP32?

See:

`Library Manager: TFT_eSPI@2.5.33 has been installed!`

```
In file included from .pio/libdeps/nodemcuv2/TFT_eSPI/TFT_eSPI.h:979,
                 from src/main.cpp:27:
.pio/libdeps/nodemcuv2/TFT_eSPI/Extensions/Smooth_font.h:47:23: warning: 'SPIFFS' is deprecated: SPIFFS has been deprecated. Please consider moving to LittleFS or other filesystems. [-Wdeprecated-declarations]
   47 |   fs::FS   &fontFS  = SPIFFS;
      |                       ^~~~~~
In file included from .pio/libdeps/nodemcuv2/TFT_eSPI/Processors/TFT_eSPI_ESP8266.h:35,
                 from .pio/libdeps/nodemcuv2/TFT_eSPI/TFT_eSPI.h:103,
                 from src/main.cpp:27:
/home/frank/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/FS.h:286:15: note: declared here
  286 | extern fs::FS SPIFFS __attribute__((deprecated("SPIFFS has been deprecated. Please consider moving to LittleFS or other filesystems.")));
```

**Comment by Bodmer** [comment]
LittleFS is already supported. SPIFFS support is also present to support legacy sketches, ignore the warning.

**Comment by hasenradball** [comment]
@Bodmer Hi,

what is not quite clear to me.

From my understanding the warning occurs due to the SPIFFS is used, right?
    => So you mean it is possible to use LittleFS but actually SPIFFS is used and due to this the warning occurs?
    => So do I switch to LittleFS manually?

From my understanding LittleFS is faster than SPIFFS.
So why not , just as question, switch to LittleFS in general and this guy who wants to use SPIFFS can then switch back to SPIFFS.

Thanks in advance

**Comment by hasenradball** [comment]
@Bodmer
In my libs when I use Filesystem I am able to switch between LittleFS and SPIFFS by doing this in the head in the include.
So in further code parts I do not need to change anything.


Kind of this:

*.h file
```
#ifndef _FS_FUNCTIONS_H_
#define _FS_FUNCTIONS_H_
#include <Arduino.h>

#ifdef ESP8266
  #include "LittleFS.h"
  #define FILESYSTEM LittleFS
  void listDir(void);
#endif

#ifdef ESP32
  #define FORMAT_SPIFFS_IF_FAILED true
  #include "FS.h"
  #include "LittleFS.h"
  #define FILESYSTEM LittleFS
  void listDir(fs::FS &fs, const char * dirname = "/", uint8_t levels = 0);
#endif
```
And in the *.cpp file usinfg `FILESYSEM` Tag like:

```
#ifdef ESP8266
void listDir() {
  DBG__PRINT(F(">>> Starte Filesystem..."));
  if (FILESYSTEM.begin()) {
    FS_isMount = true;
    DBG__PRINT(F("\tfilesystem mounted successfully!\n"));

    Dir dir = FILESYSTEM.openDir("/");
    unsigned int total{0}, fsize{0};
    while (dir.next()) {
      DBG__PRINT("\tFILE: ", dir.fileName(), " -");
      if (dir.fileSize()) {
        File fh = dir.openFile("r");
        fsize = fh.size();
        total += fsize;
        DBG__PRINT("", fsize / 1000.0F, " kB\n");
      }
      else {
        DBG__PRINT("0 kB\n");
      }
    }
    DBG__PRINT("Total size: ", total / 1000.0F, " kB\n\n");
  }
  else DBG__PRINT(F("\tERROR: filesystem mounted failed!\n\n"));
}
```
And with this you could switch between SPIFFS and LITTLEFS in the include.
Just here switched between ESP32 and ESP8266, but also possible to use this to Switch between SPIFFS and LITTLEFS.
Best regards
Frank

**Comment by hasenradball** [comment]
@Bodmer 
I think the `warning` comes from this file:

`TFT_eSPI_ESP8266.h` line 31 to 37....

```
// If smooth fonts are enabled the filing system may need to be loaded
#ifdef SMOOTH_FONT
  // Call up the SPIFFS FLASH filing system for the anti-aliased fonts
  #define FS_NO_GLOBALS
  #include <FS.h>
  #define FONT_FS_AVAILABLE
#endif
```

Did you miss to switch to LITTLE FS here?
Or is this the reason why you don' t switch to LittleFS?

> There are two filesystems for utilizing the onboard flash on the ESP8266: SPIFFS and LittleFS.
> SPIFFS is the original filesystem and is ideal for space and RAM constrained applications that utilize many small files and care about static and dynamic wear levelling and don’t need true directory support. Filesystem overhead on the flash is minimal as well.
